### ISSUE TITLE
Serve static files from a specified base directory

### ISSUE THREAD

**Issue by apfong** [UCLA-CS130/Mr.-Robot-et-al.#29]
From the spec:
> For example, if you have a base filesystem directory of “/foo/bar” and using “/static” as the root URL path for static serving, then the URL “http://host:port/static/somefile.html” would serve “/foo/bar/somefile.html” to the user.

>The files should display correctly for common file extensions. For example, “.html” should show up in the browser as an HTML file, and “.jpg” should display as an image. To do this, you need to set the Content-Type HTTP response header properly.

Our server should also return a 404 response code if the file isn't found.

**Comment by kfrankc** [comment]
Pseudocode for [request_handler](http://www.boost.org/doc/libs/1_49_0/doc/html/boost_asio/example/http/server/request_handler.cpp)
```
request_handler(req, rep) {
	// Decode url to path
	url_decode(req.uri, request_path)

	// Determine the file extension

	// Open the file to send back
	full_path = root + request_path

	// Fill out the reply to be sent to client
	// ***has to do with rep.headers
}
```

**Comment by KyCodeHuynh** [comment]
The HTTP client request will look something like this: 

```
GET /file.jpg HTTP/1.1
Accept: */*
Accept-Encoding: gzip, deflate
Connection: keep-alive
Host: localhost:8081
User-Agent: HTTPie/0.9.8
```

Where that first slash refers to the root of where files are being served from.

**Comment by KyCodeHuynh** [comment]
Finally implemented by PRs #42 and #46!
